File: btypes_primitives/src/main/java/de/hhu/stups/btypes/BStruct.java
package de.hhu.stups.btypes;

public abstract class BStruct implements BObject {

	protected BStruct() {
	}

	public BBoolean isRecord() {
		return BBoolean.TRUE;
	}

	public BBoolean isNotRecord() {
		return BBoolean.FALSE;
	}
}

File: btypes_primitives/src/main/js/de/hhu/stups/btypes/BStruct.ts
import {BObject} from "./BObject";
import {BBoolean} from "./BBoolean";

export class BStruct implements BObject {

    isRecord(): BBoolean {
        return new BBoolean(true);
    }

    isNotRecord(): BBoolean {
        return new BBoolean(false);
    }
}
